src/transformer/index.ts

```diff
@@ -0,0 +1,39 @@
+import {ConfigOutput, Platform} from '../types/config-output';
+import {ConfigVersionDocument} from '../types/config-version-document';
+import merge = require('lodash.merge');
+
+export class Transformer {
+  static #PLATFORMS: Platform[] = ['ios', 'android'];
+
+  transform(versionString: string, versionDocument: ConfigVersionDocument) {
+    const timestamp = new Date().toISOString();
+    const defaultConfig = versionDocument.environments.default;
+    const environments = Object.keys(versionDocument.environments).filter(e => e !== 'default');
+
+    const output: ConfigOutput = {};
+
+    environments.forEach(env => {
+      const envConfig = versionDocument.environments[env];
+      output[env] = {};
+      Transformer.#PLATFORMS.forEach((plt: Platform) => {
+        output[env][plt] = {
+          // set the overall platform property to the env-specific platform, if it exists, or the default
+          platform:
+            envConfig[plt] && envConfig[plt].platform
+              ? envConfig[plt].platform
+              : defaultConfig[plt].platform,
+
+          // merge the default and env-specific config objects using lodash.merge; append version string and timestamp
+          config: merge({}, defaultConfig[plt], envConfig[plt], {
+            version: versionString,
+            lastUpdated: timestamp,
+          }),
+        };
+        // delete the platform key from the config object
+        delete output[env][plt].config.platform;
```

Comment:
Don't think you need to say 'what' you're doing here. However, saying 'why' would be useful. Is it not needed? What if it were left in there, would things break?

A better approach I think, would be not to include this in the first instance. So, maybe have an exclude list in the `forEach()`. This would make removing more elements much easier and is clear to maintainers from the outset that they are not needed.

Reply:
I've re-written the comment to explain why the deletion is necessary. It's a bit of a funny one but it's to do with the fact that `platform` goes at the top level of the output object, where the rest of the config sits inside a `config` object (to make signing and verifying easier)

Reply:
It helps that you have a test for this, However, I'm going to disagree on this one sorry. Unless you need the field in the data structure - while it's being built, but not in the output - there's no reason for not excluding it upfront. If you do that, your intent is much clearer and you have a place to exclude more elements should you need to. What's more, your test is still valid.